### PR TITLE
Expose public API Telemetry

### PR DESCRIPTION
**Pull request by tobrun**
Closes https://github.com/mapbox/mapbox-gl-native/issues/11493

### CODE REVIEW

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/Telemetry.java [hunk 1]
```diff
@@ -30,10 +30,20 @@ public static void initialize() {
   }
 
   private static class EventsHolder {
```
Comment (LukasPaczos) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/Telemetry.java]: Can we rename this one to `TelemetryHolder`?

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/Telemetry.java [hunk 2]
```diff
@@ -30,10 +30,20 @@ public static void initialize() {
...
-    private static final Events INSTANCE = new Events();
+    private static final Telemetry INSTANCE = new Telemetry();
   }
 
   static MapboxTelemetry obtainTelemetry() {
     return EventsHolder.INSTANCE.telemetry;
   }
+
+  public static void enable(){
```
[the remark this answers is not in the document]
Comment (tobrun) [reply platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/Telemetry.java]: That is for opt'ing out as an end user, not the developer, the previous telem integration provided a way to disable telem programmatically.

[the remark this answers is not in the document]
Comment (tobrun) [reply platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/Telemetry.java]: Capturing from internal convo that disabling can be done through a manifest tag, will remove the api.

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/Mapbox.java [hunk 3]
```diff
@@ -46,7 +46,8 @@ public static synchronized Mapbox getInstance(@NonNull Context context, @NonNull
       Context appContext = context.getApplicationContext();
       INSTANCE = new Mapbox(appContext, accessToken);
 
-      Events.initialize();
+      Telemetry.initialize();
+      Telemetry.updateDebugLoggingEnabled(true);
```
Comment (Guardiola31337) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/Mapbox.java]: Are we enabling logging by default?

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/Telemetry.java [hunk 4]
```diff
@@ -2,11 +2,12 @@
 
 
 import com.mapbox.android.telemetry.MapboxTelemetry;
+import com.mapbox.android.telemetry.SessionInterval;
 import com.mapbox.android.telemetry.TelemetryEnabler;
 import com.mapbox.mapboxsdk.BuildConfig;
 import com.mapbox.mapboxsdk.Mapbox;
 
-public class Events {
+public class Telemetry {
```
Comment (Guardiola31337) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/Telemetry.java]: Out of curiosity, why did we change the class name?

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/Mapbox.java (hunk 3 above)
Comment (tobrun) [reply platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/Mapbox.java]: this should have gone into the testapp not the sdk

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/Telemetry.java (hunk 4 above)
Comment (tobrun) [reply platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/Telemetry.java]: Events doesn't expose the purpose of the class in context of the Maps sdk, it's too generic, Telemetry is clear and straightforward.

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/Mapbox.java (hunk 3 above)
Comment (Guardiola31337) [reply platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/Mapbox.java]: What do you mean? This is the SDK 🤔

**Pull request merged**